clients/src/test/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtilsTest.java

```diff
@@ -132,9 +133,9 @@ public void testCompressDecompress(CompressionType compressionType) throws IOExc
         } else {
             assertArrayEquals(testString, compressed);
         }
-
         ByteBuffer decompressed = ClientTelemetryUtils.decompress(compressed, compressionType);
+        byte[] actualResult = Utils.toArray(decompressed, testString.length);
```

Comment:
`byte[] actualResult = Utils.toArray(decompressed);`

we should not set the size in order to check the `flip` (i.e the return buffer is ready to read)

Reply:
Got it, thanks!